taichi/analysis/verify.cpp

```diff
@@ -0,0 +1,54 @@
+#include "taichi/ir/ir.h"
+#include <vector>
+#include <unordered_set>
+
+TLANG_NAMESPACE_BEGIN
+
+class IRVerifier : public IRVisitor {
```

Comment:
Maybe we want `BasicStmtVisitor`? So that stmts inside `IfStmt` or `WhileStmt` can be verified.

Reply:
We want to add more checks involving operands, so I'll override visitor functions for all container statements.

Reply:
Oh I realized that they are implemented in the visitor of `Block`. So we can use `BasicStmtVisitor` directly.